tfx/components/example_gen/utils.py

```diff
@@ -214,9 +223,39 @@ def _glob_to_regex(glob_pattern: Text) -> Text:
   return regex_pattern
 
 
+def _verify_split_pattern_specs(
+    split: example_gen_pb2.Input.Split) -> Tuple[bool, bool, bool]:
+  """Verify and identify specs to be matched in split pattern."""
+  is_match_span = SPAN_SPEC in split.pattern
+  is_match_date = any(spec in split.pattern for spec in DATE_SPECS)
+  is_match_version = VERSION_SPEC in split.pattern
+
+  if [is_match_span, is_match_date].count(True) > 1:
+    raise ValueError('Either span spec or date specs must be specified '
+                     'exclusively.')
+
+  if is_match_span and split.pattern.count(SPAN_SPEC) != 1:
+    raise ValueError('Only one %s is allowed in %s' % (SPAN_SPEC,
+                                                       split.pattern))
+  elif is_match_date and not all(split.pattern.count(spec) == 1
+                                 for spec in DATE_SPECS):
+    raise ValueError('Exactly one of each date spec is required in %s' %
```

Comment:
nit, add YYYY, MM, DD to the error string, which helps the users to fix their config

Reply:
updated